src/context.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { TinyEmitter } from 'tiny-emitter';
+
+interface PortalProviderProps {}
+
+export interface PortalValue {
+  subscribeToPortal: Function;
+  unsubscribeFromPortal: Function;
+  setPortal: Function;
+  getPortal: Function;
+}
+
+const PortalContext = React.createContext<PortalValue | null>(null);
+
+export class PortalProvider extends React.PureComponent<PortalProviderProps> {
+  private emitter: TinyEmitter;
+  private portals: Map<string, React.ReactNode>;
+
+  constructor(props: PortalProviderProps) {
+    super(props);
+
+    this.emitter = new TinyEmitter();
+    this.portals = new Map();
```

Comment:
Actually now that you've done this, the same thing applies to this w.r.t. using a function component. 😬